### PR TITLE
Added `Counter` example, README for proofs

### PR DESCRIPTION
**Pull request by palinatolmach**
This PR 
- adds a README file with proofs description
- changes `verbose` to `false` so the default output is more digestible
- adds `CounterProof` as an example of a failing proof output

### CODE REVIEW

File: test/kontrol/README.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    VV1_inLuck_114b9705 = 1
+    ORIGIN_ID = 645326474426547203313410069153905908525362434350
+    CALLER_ID = 645326474426547203313410069153905908525362434350
+    TIMESTAMP_CELL = 0
+    VV0_x_114b9705 = 12648430
+    NUMBER_CELL = 0
+```
+The path condition tells us that the failure occurs when `x == 12648430` and `inLuck == 1`, where `12648430` is the decimal representation of `0xC0FFEE`, and `1` represents `true` for Boolean values.
+
+The model represents concrete assignments to symbolic variables---`x`, `inLuck`, globally accessible variables such as `msg.sender` (`CALLER_ID`) and `tx.origin` (`ORIGIN_ID`)---that lead to the failure.
```
Comment (anvacaru) [inline test/kontrol/README.md]: typo?
```suggestion
The model represents concrete assignments to symbolic variables `x`, `inLuck`, globally accessible variables such as `msg.sender` (`CALLER_ID`) and `tx.origin` (`ORIGIN_ID`)---that lead to the failure.
```

File: test/kontrol/README.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+This contract demonstrates how Kontrol proofs can be used to verify properties of existing contracts, such as the ERC20 implementation from the OpenZeppelin library.
+The specification is implemented in the form of Foundry tests, however, running these tests with Kontrol ensures that function behaviors are checked for all possible input and storage values (the latter is ensured by the use of `kevm.symbolicStorage(address(dToken))` cheatcode in `setUp()`).
+
+To run these proofs, execute:
+```shell
+kontrol prove --mt GLDTokenTest.test
+```
+
+To make formal reasoning more efficient, these tests are using lemmas provided in the [lemmas.k](../../lemmas.k) file. For more information on lemmas, please refer to [our documentation](https://docs.runtimeverification.com/kontrol/guides/advancing-proofs).
```
Comment (anvacaru) [inline test/kontrol/README.md]: this should be updated with the latest file. 
```suggestion
To make formal reasoning more efficient, these tests are using lemmas provided in the [lemmas.md](../../lemmas.k) file. For more information on lemmas, please refer to [our documentation](https://docs.runtimeverification.com/kontrol/guides/advancing-proofs).
```

File: test/kontrol/README.md (hunk 1 above)
Comment (palinatolmach) [reply test/kontrol/README.md]: I was trying to say that globally accessible variables are symbolic too, but I agree that it doesn't read very well :)

File: test/kontrol/README.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
+# Euler Vault Kit Kontrol Tests
+-------------------------
+
+In this repo you'll find a series of examples showcasing some of the features of Kontrol, an open-source tool for formal verification of Solidity smart contracts. 
+
+## Getting Kontrol
+
+We recommend installing Kontrol via the K framework package manager, `kup`.
```
Comment (JuanCoRo) [inline test/kontrol/README.md]: I'd add, after the period, " We also recommend using the latest version of Foundry."

File: test/kontrol/README.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+This command will ensure that `DToken` contract's `allowance` function returns zero for _any_ input addresses.
+
+### Examples
+
+In this repository, you can find the following examples:
+
+#### 1. [Counter](./proofs/Counter.k.sol)
+
+Taken from the [Kontrol documentation](https://docs.runtimeverification.com/kontrol/guides/kontrol-example/property-verification-using-kontrol), `CounterTest` is a simple (but eloquent) example of how Kontrol can catch an issue missed by a fuzzer.
+Please note that, at the moment, this example is not available in the Docker container.
```
Comment (JuanCoRo) [inline test/kontrol/README.md:68]: Should we remove this and just re-create the docker image after this PR is merged?

File: test/kontrol/README.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    VV1_inLuck_114b9705 = 1
+    ORIGIN_ID = 645326474426547203313410069153905908525362434350
+    CALLER_ID = 645326474426547203313410069153905908525362434350
+    TIMESTAMP_CELL = 0
+    VV0_x_114b9705 = 12648430
+    NUMBER_CELL = 0
+```
+The path condition tells us that the failure occurs when `x == 12648430` and `inLuck == 1`, where `12648430` is the decimal representation of `0xC0FFEE`, and `1` represents `true` for Boolean values.
+
+The model represents concrete assignments to symbolic variables (`x`, `inLuck`, globally accessible variables such as `msg.sender` (`CALLER_ID`) and `tx.origin` (`ORIGIN_ID`)) that lead to the failure.
```
Comment (JuanCoRo) [inline test/kontrol/README.md]: I'd maybe rephrase this along the following lines, but feel free to leave it as is!
> The model represents concrete assignments to the symbolic variables `x` and `inLuck` that lead to the failure. It also concretizes globally accessible variables such as `msg.sender` (`CALLER_ID`) and `tx.origin` (`ORIGIN_ID`), although their values are not relevant for this example.

File: test/kontrol/README.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+```
+
+#### 4. [EVault](./proofs/EVaultProof.k.sol)
+
+`EVaultProof` showcases the external computation feature of Kontrol. Kontrol is both time and resource intensive, but here's a way of saving an arbitrary amount of time when executing a proof in Kontrol.
+This example instructs on how to offload the initial part of your proof computation to Foundry (which is blazing fast) and then incorporate it into a Kontrol proof.
+
+The way we achieve this is via the cheatcodes `vm.stopAndReturnStateDiff` or `vm.dumpState` (undocumented at the time of writing).
+Depending on the cheatcode used the approach is slightly different. However, in both cases we produce a JSON containing the state of the chain after executing some code, and load that state into Kontrol.
+In this example, we are using `vm.dumpState` since it's easier to work with. For complete instructions on how to include external computation with `vm.stopAndReturnStateDiff`, please see the [documentation](https://github.com/runtimeverification/kontrol/tree/master/docs/external-computation).
```
Comment (JuanCoRo) [inline test/kontrol/README.md]: I plan to upstream the documentation for `vm.dumpState` into Kontrol either today or tomorrow, so maybe we can delete the mention of undocumentation :slightly_smiling_face:

File: test/kontrol/README.md (hunk 3 above)
Comment (palinatolmach) [reply test/kontrol/README.md]: Done!

File: test/kontrol/README.md (hunk 5 above)
Comment (palinatolmach) [reply test/kontrol/README.md]: Sounds great, thanks for the suggestion!

File: test/kontrol/README.md (hunk 6 above)
Comment (palinatolmach) [reply test/kontrol/README.md]: Done, thanks!

File: test/kontrol/README.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+- Run `forge script test/kontrol/init-state/InitState.sol:InitState --sig runSetUp --ffi` to execute this function and generate JSON files ([`init-state/InitState.json`](./init-state/InitState.json) and [`init-state/AddressNames.json`](./init-state/AddressNames.json)) with state updates corresponding to the deployment process and saved addresses. 
+- Generate helper contracts, converting the generated JSON to a Solidity contract. This is done by running
+```shell
+kontrol load-state InitState test/kontrol/init-state/InitState.json --contract-names test/kontrol/init-state/AddressNames.json --output-dir test/kontrol/proofs/utils`.
+```
+This will generate two Solidity contracts in the [proofs/utils](./proofs/utils) directory that can be used to load the state into Kontrol.
+- You can then load the state into Kontrol by implementing a proof contract that inherits the [proofs/utils/InitState.sol](./proofs/utils/InitState.sol) contract generated in the previous step:
```
Comment (JuanCoRo) [inline test/kontrol/README.md]: I think this is incorrect since inheriting from `InitState` only gives access to the names of the addresses. To load the state, you have to either run the `recreateState` function or prove with the `--init-state-from-dump`. So maybe rephrase this as something like:
> Write a proof contract that inherits [proofs/utils/InitState.sol](./proofs/utils/InitState.sol), generated in the previous step, to access the names of the deployed contracts. Note that the generated contract has a `recreateState` function to recreate the state updates recorded with Foundry. However, it is faster to add said state updates through the `kontrol prove` command as instructed below.

File: test/kontrol/README.md (hunk 4 above)
Comment (palinatolmach) [reply test/kontrol/README.md:68]: Yeah, that's an option too! I wasn't sure if we'll make it today (I'll need to be afk for a couple of hours) and I've been meaning to send a follow-up message with it sooner than later... maybe we can send it as is, and then I'll update the PR once the Docker is re-built?

File: test/kontrol/README.md (hunk 7 above)
Comment (palinatolmach) [reply test/kontrol/README.md]: Done!
